ministryofjustice/analytics-platform-ops
# Prometheus Operator values

## What
Values used when deploying the kube-prometheus release with `helm`.

[Trello](https://trello.com/c/1Ced1fHi)

We decided to stop using the stable Prometheus chart in favour of Coreos's Prometheus-Operator.  The operator offers benefits such as k8s resources to easily scrape metrics from applications and create alerts.

Also, there has been talk of replacing the default stable chart with the operator.

## How to review

1. Add a `ServiceMonitor` to an application running on the cluster.

For example:
```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: my-webapp
  labels:
    app: my-webapp
    prometheus-enabled: "true"
spec:
  endpoints:
    - port: 80
       interval: 30s
       honorLabels: true
  selector:
    matchLabels:
      app: my-webapp
  namespaceSelector:
    matchNames:
    - user-default
```

2. Deploy it:  `kubectl apply -f my-webapp-servicemonitor.yaml`

3. You should now be able to see metrics for your application in the Prometheus ui.

Helpful [link](https://coreos.com/operators/prometheus/docs/latest/user-guides/getting-started.html)
